src/models/release.ts

```diff
@@ -729,6 +729,204 @@ const getReleaseModel = function (
+		 * @name upload
+		 * @public
+		 * @function
+		 * @memberof balena.models.release.asset
+		 *
+		 * @param {Object} uploadParams - upload parameters
+		 * @param {String|File} uploadParams.asset - asset file path (string, Node.js only) or File object (Node.js & browser). For File objects, use new File([content], filename, {type: mimeType})
+		 * @param {String} uploadParams.asset_key - unique key for the asset within the release
+		 * @param {Number} uploadParams.release - release ID
+		 * @param {Object} [options={}] - upload options
+		 * @param {Number} [options.chunkSize=5242880] - chunk size for multipart uploads (5MiB default)
+		 * @param {Number} [options.parallelUploads=5] - number of parallel uploads for multipart
+		 * @param {Boolean} [options.overwrite=false] - whether to overwrite existing asset
+		 * @param {Function} [options.onUploadProgress] - callback for upload progress
+		 * @fulfil {Object} - uploaded release asset
+		 * @returns {Promise}
+		 *
+		 * @example
+		 * // Upload from file path (Node.js)
+		 * balena.models.release.asset.upload({
+		 * 	asset: '/path/to/logo.png',
+		 * 	asset_key: 'logo.png',
+		 * 	release: 123
+		 * }).then(function(asset) {
+		 * 	console.log('Asset uploaded:', asset);
+		 * });
+		 *
+		 * @example
+		 * // Upload with File API (Node.js and browser)
+		 * const content = Buffer.from('Hello, World!', 'utf-8');
+		 * const file = new File([content], 'readme.txt', { type: 'text/plain' });
+		 *
+		 * balena.models.release.asset.upload({
+		 * 	asset: file,
+		 * 	asset_key: 'readme.txt',
+		 * 	release: 123
+		 * }).then(function(asset) {
+		 * 	console.log('Asset uploaded:', asset);
+		 * });
+		 *
+		 * @example
+		 * // Upload large file with File API and progress tracking
+		 * const largeContent = new Uint8Array(10 * 1024 * 1024); // 10MB
+		 * const largeFile = new File([largeContent], 'data.bin', { type: 'application/octet-stream' });
+		 *
+		 * balena.models.release.asset.upload({
+		 * 	asset: largeFile,
+		 * 	asset_key: 'data.bin',
+		 * 	release: 123
+		 * }, {
+		 * 	chunkSize: 5 * 1024 * 1024, // 5MB chunks
+		 * 	parallelUploads: 3,
+		 * 	onUploadProgress: function(progress) {
+		 * 		const percent = (progress.uploaded / progress.total * 100).toFixed(2);
+		 * 		console.log(`Upload progress: ${percent}%`);
+		 * 	}
+		 * }).then(function(asset) {
+		 * 	console.log('Large file uploaded:', asset);
+		 * });
+		 *
+		 * @example
+		 * // Browser: Upload file from input element
+		 * const fileInput = document.getElementById('fileInput');
+		 * const file = fileInput.files[0]; // File object from input
+		 *
+		 * balena.models.release.asset.upload({
+		 * 	asset: file,
+		 * 	asset_key: file.name,
+		 * 	release: 123
+		 * }).then(function(asset) {
+		 * 	console.log('File uploaded from browser:', asset);
+		 * });
+		 *
+		 * @example
+		 * // Upload with overwrite option
+		 * balena.models.release.asset.upload({
+		 * 	asset: '/path/to/logo.png',
```

Comment:
Let's put the two examples that are using a string path next to each other since they are node only